packages/wallet/ui/src/components/tests/Offer.test.jsx

```diff
@@ -128,6 +128,63 @@ test('renders the wants', () => {
   );
   expect(want.text()).toContain('into Zoe fees');
 });
+
+test('renders unknown amounts when parsing fails', () => {
+  const invalidOffer = {
+    status: 'proposed',
+    id: '123',
+    offerId: 'https://tokenpalace.app#555',
+    instancePetname: ['TokenPalace', 'Installation'],
+    instanceHandleBoardId: '123',
+    actions: {
+      accept: jest.fn(),
+      cancel: jest.fn(),
+      decline: jest.fn(),
+    },
+    requestContext: {
+      date: 1636614038901,
+      dappOrigin: 'https://tokenpalace.app',
+      origin: 'unknown origin',
+    },
+    proposalForDisplay: {
+      arguments: [],
+      give: {
+        Collateral: {
+          amount: {
+            brand: { kind: 'brand', petname: 'PAL' },
+            displayInfo: { assetKind: 'nat', decimalPlaces: 6 },
+          },
+          pursePetname: 'Palace Token',
+        },
+      },
+      want: {
+        RUN: {
+          amount: {
+            brand: { kind: 'brand', petname: 'RUN' },
+            displayInfo: { assetKind: 'nat', decimalPlaces: 6 },
+            value: 18544198n,
+          },
+          pursePetname: 'Zoe fees',
+        },
+      },
+    },
+    invitationDetails: {
+      fee: {
+        value: 5000000n,
+        brand: { kind: 'brand', petname: 'RUN' },
+        displayInfo: { assetKind: 'nat', decimalPlaces: 6 },
+      },
+      feePursePetname: 'Zoe fees',
+      expiry: 1723014088n,
+    },
+  };
+  const component = mount(<Offer offer={invalidOffer} />);
+
+  const give = component.find('.OfferEntry').at(0);
+
+  expect(give.text()).toContain('unknown');
```

Comment:
is this accurate? this appears to the user like the system doesn't know the value. the system does but the UI encountered an error rendering it. We'd want bug reports in that case? If so, something like `<error>` would be better.

Reply:
I meant simply `toContain('<error>')` as the string. an ErrorBoundary is a good idea though, to make the app more robust. I think this particular parsing error made the app crash.